tests/acceptance/DownloadGigaDBTemplateFiles.feature

```diff
@@ -0,0 +1,22 @@
+Feature: An author can download file template
+  As an author
+  I want to download upload form template after login
+  So that I can upload data using the form
+
+  Background: Sign is as a user
+    Given I sign in gigadb website as "user"
+    Then I should see "John's GigaDB Page"
+
+  @ok
+  Scenario: Go to Download template file page as a user and each download button linked with correct file path
+    When I am on "/datasetSubmission/upload"
+    Then I should see a link "Download Template File" to "/files/templates/GigaDBUploadForm-forWebsite-v22Dec2021.xlsx"
+    And I should see a link "Download Example File 1" to "/files/templates/GigaDBUpload-Example1-forWebsite-v22Dec2021.xlsx"
+    And I should see a link "Download Example File 2" to "/files/templates/GigaDBUpload-Example2-forWebsite-v22Dec2021.xlsx"
+
+  @ok
+  Scenario: Download template file as a user
+    When I am on "/datasetSubmission/upload"
+    Then I press "Download Template File" and the response should contain "200"
+    And I press "Download Example File 1" and the response should contain "200"
+    And I press "Download Example File 2" and the response should contain "200"
```

Comment:
Each of these 3 steps read like they should be 2 steps:
```
And I press "Download Template File"
Then the response should contain "200"
```

This is what is done in [`dataset-view.feature`](https://github.com/gigascience/gigadb-website/blob/develop/features/dataset-view.feature) but I think I understand why you combined the 2 steps because it does not look like there is an easy way to test HTTP response code in a separate step in Codeception like you can do in Behat acceptance tests.

I guess you could test for actual file download in an acceptance test - see [here](https://stackoverflow.com/questions/32828524/codeception-how-do-i-test-file-download). Maybe @rija has another solution.

Reply:
Hi @kencho51,

@pli888 makes good point here,

I'd actually go further saying that in addition to that, you should make a separate scenario for each template file (I think I made a similar comment on another of your PR)
as they are different things that although are treated identically for now but that might change in the future.
We don't want to have to meddle with an existing and passing test scenario when a new template comes in or one of the template needs to be treated differently.

I think pressing a button and checking the files exists locally is a reasonable approach.

Testing for HTTP response is more of functional/integration tests level. Acceptance tests are from a user/product owner perspective and they don't care much about http response.

Reply:
Hi @rija , 

I tried to test for actual file download, first by specifying the download path in `acceptance.suite.yml` and using `file_exist()` for checking.
But I found out that, the download action cannot be triggered by `I press the button "Download Example File 1"` as no file is downloaded. I tried to explicitly `I press the button "http://gigadb.test/files/templates/GigaDBUpload-Example1-forWebsite-v22Dec2021.xlsx"`, but got `invalid CSS and XPath selector and Link or Button` error. 
Can you have a look for me?

Reply:
I've not been able to get file download working acceptance tests today. Maybe the `selenium/standalone-chrome:3.141.59-oxygen` image in `docker-compose.yml` could do with  being updated to a newer version.